src/main/kotlin/uk/gov/gdx/datashare/services/GroApiService.kt

```diff
@@ -4,18 +4,34 @@ import com.fasterxml.jackson.databind.ObjectMapper
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Service
+import uk.gov.gdx.datashare.config.DateTimeHandler
 import uk.gov.gdx.datashare.config.NoDataFoundException
 import uk.gov.gdx.datashare.models.GroDeathRecord
+import uk.gov.gdx.datashare.models.GroDeleteEventResponse
 import uk.gov.gdx.datashare.models.GroEnrichEventResponse
+import uk.gov.gdx.datashare.repositories.SupplierEventRepository
+import java.util.*
 
 @Service
 class GroApiService(
+  private val dateTimeHandler: DateTimeHandler,
   private val lambdaService: LambdaService,
   private val objectMapper: ObjectMapper,
-  @Value("\${enrich-event-lambda-function-name:#{null}}") val functionName: String?,
+  private val supplierEventRepository: SupplierEventRepository,
+  @Value("\${delete.event.lambda.function.name:#{null}}") val deleteFunctionName: String?,
+  @Value("\${enrich-event-lambda-function-name:#{null}}") val enrichFunctionName: String?,
 ) {
+  fun deleteConsumedGroSupplierEvents() {
+    val now = dateTimeHandler.now()
+    supplierEventRepository.findGroDeathEventsForDeletion()
+      .forEach {
```

Comment:
discussed breaking this down in the scheduler so that the transaction wraps each item rather than the list

Reply:
Working on it